src/js/components/day.jsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import * as moment from 'moment';
+import Timeslot from 'timeslot';
+
+import {
+  DEFAULT_TIMESLOT_FORMAT,
+  DEFAULT_TIMESLOT_SHOW_FORMAT,
+  DEFAULT_TIMESLOTS,
+} from 'constants/day.js';
+
+export default class Day extends React.Component {
+  render() {
+    const dayClassNames = classnames({
+      'tsc-day': true,
+    });
+
+    const {
+      timeslotFormat,
+      timeslotShowFormat,
+      timeslots,
+      onClick,
+
+    } = this.props;
+
+    const slots = timeslots.map((slot, index) => {
```

Comment:
This function is too long, make a private function `_renderTimeSlots()` to handle this.